playground/context/excalidraw.ts

```diff
@@ -0,0 +1,28 @@
+import type { ExcalidrawElementSkeleton } from "@excalidraw/excalidraw/types/data/transform";
+import type {
+  BinaryFiles,
+  ExcalidrawImperativeAPI,
+} from "@excalidraw/excalidraw/types/types";
+import React, { useContext } from "react";
+import { parseMermaid } from "../../src/parseMermaid";
+
+export const ExcalidrawContext = React.createContext<{
+  excalidrawAPI?: ExcalidrawImperativeAPI;
+  addFiles: (files: BinaryFiles) => void;
+  updateElements: (elements: ExcalidrawElementSkeleton[]) => void;
+  translateMermaidToExcalidraw: (mermaidSyntax: string) => Promise<{
+    mermaid: Awaited<ReturnType<typeof parseMermaid>>;
+    excalidraw: { elements: ExcalidrawElementSkeleton[]; files?: BinaryFiles };
+  }>;
+  setApi: (api: ExcalidrawImperativeAPI) => void;
+} | null>(null);
+
+export const useExcalidraw = () => {
+  const context = useContext(ExcalidrawContext);
+
+  if (!context) {
+    throw new Error("useExcalidraw must be used within a ExcalidrawProvider");
+  }
+
+  return context;
+};
```

Comment:
@ad1992 I honestly don't like having the state in the App, do you know if it makes sense to just keep the Excalidraw API in the App?

I couldn't think of a way to have this state without using context.

Reply:
Not really unnecessary rerenders :) It depends on how the component is structured and since this is a very simple playground app so context seems unnecessary here hence I suggested to move it to App level instead